File: MODEL/punto.cpp
#include "punto.h"
#include <cmath>

Punto::Punto():x(0),y(0){}

Punto::Punto(double xx, double yy):x(xx),y(yy){}

const Punto Punto::origine=Punto();

double Punto::getX()const{return x;}

double Punto::getY()const{return y;}

double Punto::distanceTo(const Punto& primo, const Punto& secondo){
    return sqrt(
                pow((secondo.getX()-primo.getX()),2)+
                pow((secondo.getY()-primo.getY()),2)
            );
}

Angolo Punto::angoloTraTrePunti(const Punto& A, const Punto& B, const Punto& C){
    double latoAB = distanceTo(A,B);
    double latoBC = distanceTo(B,C);
    double latoAC = distanceTo(A,C);
    double cos_a =(( pow(latoAC,2) + pow(latoAB,2) - pow(latoBC,2)) / (2*latoAB*latoAC)) ;
    return Angolo( acos(cos_a) *180/PI );
}

File: KalkJava/src/Punto.java
import java.lang.Math;

public class Punto {

    public Double x;
    public Double y;
	
    public Punto(Double x, Double y){
        this.x = x;
        this.y = y;
    }
    public Punto(){
    	this(0,0);
    }
    public Double getX() {
    	return x;
    }
    public Double getY() {
    	return y;
    }
    public void invertiY() {
    	y = -y;
    }
	public static Double distanceTo(Punto a,  Punto b) {
		 return Math.sqrt(
	                Math.pow((b.getX()-a.getX()),2)+
	                Math.pow((b.getY()-a.getY()),2)
	            );
	}
	public static Angolo angoloTraTrePunti(Punto C,  Punto A,  Punto B) {
	    Double latoAB = distanceTo(A,B);
	    Double latoBC = distanceTo(B,C);
	    Double latoAC = distanceTo(A,C);
	    Double cos_a =(( Math.pow(latoAC,2) + Math.pow(latoAB,2) - Math.pow(latoBC,2)) / (2*latoAB*latoAC)) ;
	    return new Angolo( Math.acos(cos_a) *180/Math.PI );
	}
	public static final Punto origine(){	//
		return new Punto();
	}
}
